src/ReactWrapper.js

```diff
@@ -269,8 +270,12 @@ export default class ReactWrapper {
    * @returns {ReactWrapper}
    */
   find(selector) {
-    const predicate = buildInstPredicate(selector);
-    return findWhereUnwrapped(this, predicate);
+    const complexSelector = new ComplexSelector(
+      buildInstPredicate,
+      findWhereUnwrapped,
+      childrenOfInst
+    );
```

Comment:
Ultimately it's up to the maintainers. They might see this argument as moot. But it could make maintenance easier when all selector engines follow a similar flow. Previously that was the case until here.

Additionally, it is heavier to instantiate an object for every selector. But I don't they worry too much about the perf for this lib.